dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -23,7 +23,7 @@ import shapeless.ops.hlist.{ToTraversable, Tupler}
 class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val encoder: TypedEncoder[T])
     extends TypedDatasetForwarded[T] { self =>
 
-  private implicit val sparkContext = dataset.sqlContext.sparkContext
+  private implicit val spark: SparkSession = dataset.sparkSession
```

Comment:
Spark refers to it as "spark" internally. Since it's also shorter, I thought of using the same term. I don't have any strong feeling about it ... do we think we should use "sparkSession" to "spark"?